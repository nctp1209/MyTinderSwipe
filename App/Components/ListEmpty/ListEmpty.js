import React, { Component } from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

export default class ListEmpty extends Component {
  render() {
    return (
      <View style={styles.emptyContainer}>
        <Text style={styles.titleStyle}>{this.props.title}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  emptyContainer: {
    flex: 1,
    justifyContent: "center",
    alignContent: "center",
    height: 100,
    backgroundColor: "#FFF"
  },
  titleStyle: { fontStyle: "italic", textAlign: "center" }
});

ListEmpty.propTypes = {
  title: PropTypes.string
};

ListEmpty.defaultProps = {
  title: "No data"
};
