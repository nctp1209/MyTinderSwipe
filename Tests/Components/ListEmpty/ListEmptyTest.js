import React from "react";
import renderer from "react-test-renderer";

import ListEmpty from "../../../App/Components/ListEmpty/ListEmpty";

test("ListEmpty without title", () => {
  const tree = renderer.create(<ListEmpty />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("ListEmpty with title", () => {
  const tree = renderer.create(<ListEmpty title="Test ListEmpty" />).toJSON();
  expect(tree).toMatchSnapshot();
});
