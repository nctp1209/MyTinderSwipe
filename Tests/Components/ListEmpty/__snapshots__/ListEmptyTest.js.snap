// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListEmpty with title 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "height": 100,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontStyle": "italic",
        "textAlign": "center",
      }
    }
  >
    Test ListEmpty
  </Text>
</View>
`;

exports[`ListEmpty without title 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "height": 100,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontStyle": "italic",
        "textAlign": "center",
      }
    }
  >
    No data
  </Text>
</View>
`;
